enumerable_methods.rb

```diff
@@ -0,0 +1,214 @@
+module Enumerable
+  def my_each
+    return 'No block given' unless block_given?
+
+    for x in self
+      yield(x)
+    end
+  end
+
+  def my_each_with_index
+    return 'No block given' unless block_given?
+
+    i = 0
+    for x in self
+      yield(x, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return 'No block given' unless block_given?
+
+    arr = []
+    my_each do |el|
+      arr << el if yield(el) == true
+    end
+    p arr
+  end
+
+  def my_all?(*arg)
+    i = 0
+    if block_given?
+      my_each do |el|
+        i += 1 if yield(el) == true
+      end
+    elsif arg[0] == Integer || arg[0] == Float || arg[0] == String
+      my_each do |el|
+        i += 1 if el.class == arg[0]
+      end
+    else
+      my_each do |el|
+        i += 1 if el == arg[0]
+      end
+    end
+    i == size
+  end
+
+  def my_any?(*arg)
+    i = 0
+    if block_given?
+      my_each do |el|
+        i += 1 if yield(el) == true
+      end
+    elsif arg[0] == Integer || arg[0] == Float || arg[0] == String
+      my_each do |el|
+        i += 1 if el.class == arg[0]
+      end
+    else
+      my_each do |el|
+        i += 1 if el == arg[0]
+      end
+    end
+    i.positive?
+  end
+
+  def my_none?(*arg)
+    i = 0
+    if block_given?
+      my_each do |el|
+        i += 1 if yield(el) != true
+      end
+    elsif arg[0] == Integer || arg[0] == Float || arg[0] == String
+      my_each do |el|
+        i += 1 if el.class != arg[0]
+      end
+    else
+      my_each do |el|
+        i += 1 if el != arg[0]
+      end
+    end
+    i == size
+  end
+
+  def my_count(*arg)
+    counter = 0
+    if block_given?
+      my_each do |el|
+        counter += 1 if yield(el) == true
+      end
+      counter
+    elsif arg.empty?
+      size
+    else
+      my_each do |el|
+        counter += 1 if el == arg[0]
+      end
+      counter
+    end
+  end
+end
+
+module Enumerable
+  def my_map(*_arg)
+    return 'No block given' unless block_given?
+
+    i = 0
+    arr = []
+    my_each do |el|
+      arr[i] = yield(el)
+      i += 1
+    end
+    arr
+  end
```

Comment:
```rb
2.6.5 :069 > array
 => [1, 0, 6, 4, 8, 3, 6, 2, 8, 2, 3, 6, 7, 0, 5, 0, 0, 7, 0, 7, 8, 0, 3, 3, 1, 3, 1, 8, 0, 6, 2, 5, 3, 2, 5, 7, 4, 8, 8, 6, 8, 3, 2, 0, 7, 6, 3, 3, 0, 2, 4, 0, 5, 3, 1, 7, 5, 7, 7, 6, 2, 1, 5, 0, 4, 2, 0, 5, 4, 3, 8, 2, 8, 4, 6, 8, 7, 6, 3, 7, 5, 1, 5, 8, 6, 4, 0, 1, 4, 3, 6, 1, 7, 7, 6, 5, 1, 7, 6, 0] 
2.6.5 :070 > array.my_map.is_a?(Enumerator)
 => false 
2.6.5 :071 > array.map.is_a?(Enumerator)
 => true 
2.6.5 :072 >
```

Reply:
```rb
2.6.5 :074 > my_proc = proc {|num| num > 10 }
2.6.5 :075 > array.my_map(my_proc) {|num| num < 10 } == array.map(&my_proc)
 => false 
2.6.5 :076 >
```

Reply:
- The first issue was corrected.
- On the econd one I think you mistakenly compared {|num| num > 10 } with {|num| num < 10 } haha. When you change it, it works.